engine/src/main/java/org/terasology/network/internal/NetworkSystemImpl.java

```diff
@@ -868,6 +900,18 @@ private void connectClient(Client client) {
         for (BlockFamily registeredBlockFamily : blockManager.listRegisteredBlockFamilies()) {
             serverInfoMessageBuilder.addRegisterBlockFamily(registeredBlockFamily.getURI().toString());
         }
+        Path homePath = PathManager.getInstance().getHomePath();
+        try {
+            for (String line : Files.readAllLines(homePath.resolve("blacklist.json"))) {
```

Comment:
It seems odd to me that we have class ServerConnectListManager for encapsulating blacklists and whitelists, but we're reading the data from the files rather than from ServerConnectListManager.   I get it that reading the data from the files is easier than reading it from the list collections, but moving this code into a method of ServerConnectListManager can hide that fact from the caller.   So you're going to want a method like
`
for (String line : serverConnectListManager.getWhiteListLines()) {
`
 instead of 
`
for (String line : Files.readAllLines(homePath.resolve("whitelist.json"))) {
`

When all is said and done, there shouldn't be any references to the whitelist and blacklist files outside of ServerConnectListManager.   Someone should be able to swap in a different implementation of ServerConnectListManager such as reading from a public REST service instead of local files without changing any other code.